flow/layers/clip_rect_layer.cc

```diff
@@ -41,7 +41,7 @@ void ClipRectLayer::Paint(PaintContext& context) const {
   TRACE_EVENT0("flutter", "ClipRectLayer::Paint");
   FML_DCHECK(needs_painting());
 
-  SkAutoCanvasRestore save(&context.canvas, clip_behavior_ != Clip::hardEdge);
+  SkAutoCanvasRestore save(&context.canvas, true);
```

Comment:
Ah, yes, this was my mistake.

It should be
```
SkAutoCanvasRestore save(&context.canvas, true);
context.canvas.clipRect(paint_bounds(), clip_behavior_ != Clip::hardEdge);
```

BTW, can you write a golden test in flutter/flutter to verify this bug and ensure that we won't regress in the future?

Reply:
Ahh that version makes a lot more sense! Yeah, I'll get a golden test in.